src/components/CreateArticle.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createArticle } from '../actions/articleAction';
+import { addFlashMessage } from '../actions/flashActions';
+import FlashMessagesList from './FlashMessagesList';
+import uploadImageCloudinary from '../utilities/cloudinaryUpload';
+
+class CreateArticle extends Component {
+  constructor(props) {
+    super(props);
+    const {
+      cache: {
+        id, photo, title, body, image
+      }
+    } = this.props;
+    this.state = {
+      id,
+      photo,
+      title,
+      body,
+      image,
+    };
+  }
+
+  async componentDidUpdate(prevProps, prevState) {
+    if ((this.props.publish && prevProps.publish === false)
+    || (this.props.draft && prevProps.draft === false)) {
+      let uploaded = true;
+      const { onSave, createArticle } = this.props;
+      const {
+        photo,
+        image,
+        id,
+        title,
+        body
+      } = prevState;
+      if (title.trim().length < 3 || body.trim().length < 3) {
+        onSave();
+        this.props.addFlashMessage({
+          type: 'error',
+          text: 'title and body must not be less than 3 characters'
+        });
+      } else {
+        if (photo && image) {
+          uploaded = await uploadImageCloudinary(image);
+        }
+        if (uploaded) {
+          uploaded = ((typeof uploaded) === 'string') ? uploaded : null;
+          // eslint-disable-next-line react/no-did-update-set-state
+          this.setState({ image: uploaded }, async () => {
+            if (id) {
+            //
+            } else {
+              onSave();
+              const article = await createArticle(this.state);
+              if (!image) {
+                this.onClose();
+              }
+              if (article.success) {
+                this.setState({
+                  id: article.data.id,
+                  title: article.data.title,
+                  description: article.data.description,
+                  body: article.data.body,
+                  readTime: article.data.readTime,
+                  image: article.data.image
+                });
+                this.props.addFlashMessage({
+                  type: 'success',
+                  text: 'Article saved to draft'
+                });
+              } else {
+                this.props.addFlashMessage({
+                  type: 'error',
+                  text: article.message
+                });
+              }
+            }
+          });
+        } else {
+          this.props.addFlashMessage({
+            type: 'error',
+            text: 'failed to upload image'
+          });
+        }
+      }
+    }
+  }
+
+  onChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+    const scroll = event.target.scrollHeight;
```

Comment:
Why do we need this?

Reply:
To increase the height of the text area automatically without having to scroll

Reply:
I’ll leave this for now. But I’ll try to remember to come back to it